### ISSUE TITLE
Proximity sensor won't turn off the screen during calls

### ISSUE THREAD

**Issue by xashyar** [signalapp/Signal-Android#12553]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
The proximity sensor works for normal calls, and other apps on the same phone, but not for Signal.

It even works, inside Signal for switching voice note audio outputs from speaker to earpiece.

### Steps to reproduce
- Start a call using Signal
- See the screen won't turn off

**Actual result:** Touch screen buttons get unwanted taps

**Expected result:** Screen turns off

### Device info

**Device:** Manufacturer Model XVI
**Android version:** 7.0.0
**Signal version:** 6.0.1

### Link to debug log

https://debuglogs.org/android/6.0.1/4f529e0914fb290596ec5025670a8c6bd3b685b405b0ac8332958181cd414921

**Comment by R101G** [comment]
@Xashyar,  I use a Unihertz Jelly2 with Android 11.

**Comment by Xashyar** [comment]
Annoyingly still reproducibile.

**Comment by thegeni** [comment]
Very annoying problem. Reproduced on my Oneplus 8 Pro and Google Pixel 7 Pro, both on Android 13.

**Comment by Fehertukor** [comment]
Oneplus too. Signal screen goes crazy, 
but the opposite way! 
It stays black and I have to wave before the screen (after a normal phone call) to "wake it up".
This is the same experience with and without screen protector.

**Comment by niktss** [comment]
Also on Galaxy A52s 5G, Android 13
https://debuglogs.org/android/6.34.2/beb8a843ff4a9081ee6e506a1cf7e2e9b33099446221624e92b5e4646b0ad107

**Comment by gpkvt** [comment]
+1

Fairphone 5, Android 13

**Comment by hetszunyukapanyanyimonyok** [comment]
I had this same problem. 
Just checked and all of my notifications are enabled. I will check @cody-signal solution and reply as soon as possible.

**Comment by korg91** [comment]
This bug is still present in 7.28.4 on my Pixel 8. Signal calls behave as if the proximity sensor did not exist. I tried both receiving and making calls, both with screen locked and unlocked, it makes no difference. Notifications are enabled, so that's not the issue.

**Comment by Hurbad** [comment]
Still experiencing this bug on Android 14, Nothing Phone 2A 7.28.4. The proximity sensor starts acting up.

**Comment by korg91** [comment]
Update: today I received a call and the proximity sensor worked as expected. Then I received other calls and it did not. I think it's much more frequent that it does not work. I could not understand if there's any specific factor that affects the behavior.

**Comment by bigon** [comment]
Hello,

I think that #13646 is duplicate of this one.

I can confirm on my phone that enabling/disabling the speaker workaround the issue and the proximity sensor is working as expected

**Comment by lietu** [comment]
Just had a call on Signal, phone was freaking out the whole time. Still an issue.

Pixel 6 Pro + GrapheneOS

**Comment by Ansarmughal99** [comment]
Experienced the Same issue today
Device relme gt neo 2 
A13 
It works 1 time only. Again it was not working
If i ond and off speaker during call it response sometime but that is not a soultion.

Here is the log

https://debuglogs.org/android/7.32.3/91cc0fc5bdaadac540a4399fb13cb08e4dcea597643d576dbb0ac090ef6973ae

**Comment by dimitris-personal** [comment]
I can reproduce just about on demand on up-to-date Pixel phones (6a, 7).  Up to date on OS, Play Services.

Here's a debug log from the P6a.  Phone "asleep" (screen dark/locked), Received the incoming call (initiated on the P7).  Screen lit up, was able to answer the call but proximity sensor was not having any effect.

https://debuglogs.org/android/7.32.3/284ca41c668857308a96758022aabf7b907c96bf9cfc1410dc8e885c42f86625

If it matters, I'm using Niagara launcher on both phones.

**Comment by Garbulix** [comment]
I confirm this issue.

Pixel 6a
GrapheneOS based on Android 15
Signal 7.34.2 from Google Play

**Comment by kevenwyld** [comment]
I'm also having this issue but only sometimes.  It seems as though the first time I put the phone to my ear after starting a call the proximity sensor works, but if I remove it to look at the screen and then put it back to my ear it won't turn off automatically again.

I'll see if I can get a debug log right after the issue occurs next time I notice it.

GrapheneOS (Android 15)
Signal 7.37.3 from the APK

**Comment by kevenwyld** [comment]
[Here is](https://debuglogs.org/android/7.37.3/4e3b280c4f06a1dfd388457d2a268fbb8e95707df9d0330988449b66cc9af803) the debug log.  I tested an incoming call and the behavior is slightly different than I described above.  When I received the call and answered with the phone locked the screen did not turn off when I put it on my ear, or when repeatedly moving it away from my ear and back again.

I then tried turning the screen off with the power button and back on again (phone still locked).  After that the proximity sensor worked correctly, even with multiple times of moving it away and back again it worked.

I do feel like sometimes it's the other way around, where it works once and then doesn't work again, but this is how it went with this test.

GrapheneOS (Android 15)
Signal 7.37.3 from the APK
Pixel 7

**Comment by dimitris-personal** [comment]
Still reproduces with latest Android and Signal.  Adding my comment here from my other issue (which I closed as it had drifted and at least partially duplicates this one):

_Note the possible correlation with doze mode_

Readily reproducable here on the same Pixel 7.

Incidentally, and maybe separate issue (or not?) the first call attempt failed to surface a ring (and call never went to Ringing on the originating device), but the moment I dropped the call attempt from the originating device I got the missed call notification on the P7.

On the second attempt, after accepting the call I got the "no proximity screen disable until I manually toggle screen off/on with the power button" issue again.

Debug log: https://debuglogs.org/android/7.39.3/e9aef9eeb4e5ffe70871a5e2180af9ad347be24838415349b68ddbe844eb19ea

**Comment by kevenwyld** [comment]
I'm also still experiencing this problem.  99% of my calls are on signal and it happens on every one, very consistently.

>  first call attempt incoming on a screen-off "sleeping" Pixel 7 failed to surface a ring

@dimitris-personal is there an issue for this already?  I didn't see one but if there is I'd like to follow it.  I have that problem with maybe 1 out of every 10 calls.  But what I find interesting is that it sounds a LOT like this issue https://discuss.grapheneos.org/d/11421-graphene-os-intermittently-fails-to-ring-for-incoming-calls which is actually an upstream bug (see links on that forum) impacting the Pixel 7 that google probably won't fix.  I know it's really unlikely they are related but it does make me say "hmmm"

GrapheneOS (Android 15)
Signal 7.39.4 from the APK
Pixel 7

**Comment by SebastianJL** [comment]
For me the screen turns off during calls but not when listening to a voice message and holding the phone to the ear. The volume turns down correctly though.

Android 14
Signal 7.39.4 from PlayStore
Fairphone 5

**Comment by saaste** [comment]
I have this issue as well:

Pixel 6 Pro
Android 15
Signal 7.41.2 from Google Play

[debug log](https://debuglogs.org/android/7.41.2/b61c9a1ecefacbc0430b34ed1b23c8be31bfb354bb701a2b1e4ede7c9217ea8d)

**Comment by theofficialgman** [comment]
Same issue. Pixel 7, latest stable release android.

**Comment by junx404** [comment]
Copypasted my comment from the wrong issue thread:
I also have this problem and had it for a long time, did you find a fix? Super annoying when my ear turns on airplane mode mid call.
Pixel 7a, grapheneos with sandboxed play

**Comment by OlehAhieienko** [comment]
Same on Google Pixel 8 on both stock Android 15 and 16 😿

**Comment by dougmassay** [comment]
Still happening for me on a fully updated Pixel 8 Pro. Really annoying.

**Comment by ElijahElliotP** [comment]
Pixel 9, fully up to date, same issue. Furthermore turning off the screen doesn't invoke a lock, so my ear or face "taps" and wakes the phone once it is against my face. Have to manually close the phone again, but the phone screen should remain off and it does not.

**Comment by akalankag** [comment]
Noticed this issue on Pixel 6a, Android 16

**Comment by undermethod** [comment]
Can confirm still an issue.
- Device: HMD Skyline TA-1600
- Android 15 with 2025-07 security update
- Signal version 7.51.2

**Comment by Dami6** [comment]
> Hi folks,
> 
> 
> There are a few things that currently prevent the screen from turning off (not saying it's the best ux, just that it's currently implemented this way):
> 
>     * You have video enabled
> 
>     * You have speaker phone enabled

You are right, not so smart UX decision. Let's say I am having a video call. During the call the guy that I talk, he to moves to a noisier environment. He obviously puts the phone to his ear so that he can hear me. And I experience a fantastic closeup to a piece of ear and some hair. Not very pleasant experience, I would prefer the screen to go black in this case - and go back to normal video only once the proximity sensor says so.
